src/utils/addons/prepare.js

```diff
@@ -0,0 +1,110 @@
+const chalk = require('chalk')
+
+const ADDON_VALIDATION = {
+  EXISTS: 'EXISTS',
+  NOT_EXISTS: 'NOT_EXISTS',
+}
+
+const validateExists = ({ addon, addonName, siteData, log, exit }) => {
+  if (!addon || !addon.id) {
+    log(`Add-on ${addonName} doesn't exist for ${siteData.name}`)
+    log(`> Run \`netlify addons:create ${addonName}\` to create an instance for this site`)
+    exit(1)
+  }
+}
+
+const validateNotExists = ({ addon, addonName, siteData, log, exit }) => {
+  if (addon && addon.id) {
+    log(`The "${addonName} add-on" already exists for ${siteData.name}`)
+    log()
+    const cmd = chalk.cyan(`\`netlify addons:config ${addonName}\``)
+    log(`- To update this add-on run: ${cmd}`)
+    const deleteCmd = chalk.cyan(`\`netlify addons:delete ${addonName}\``)
+    log(`- To remove this add-on run: ${deleteCmd}`)
+    log()
+    exit(1)
+  }
+}
+
+const getCurrentAddon = ({ addons, addonName }) => {
+  const addon = addons.find(addon => addon.service_path === `/.netlify/${addonName}`)
+  return addon
+}
+
+const validateCurrentAddon = ({ addon, validation, addonName, siteData, log, warn, exit }) => {
+  switch (validation) {
```

Comment:
Should this validation enum be simplified to a boolean instead?

Reply:
I started with a boolean and had a hard time naming it 🤦
The opposite of `validateExists`  can mean - `don't validate that it exists`, but it our case it means `validate that it doesn't exists`.
If you think a boolean is easier to understand I'm good with changing it

Reply:
I think either works :)
A boolean would be simpler, but an enum would more flexible to additional validation cases (but I'm not sure this might happen there).